src/dlt_mcp/_utilities/ingestion.py

```diff
@@ -0,0 +1,343 @@
+"""Utilities to ingest `dlt` documentation and code to power the MCP search tools."""
+
+import zipfile
+import tempfile
+from pathlib import Path
+from typing import Callable, Iterator
+
+import chonkie.logger
+import lancedb
+import numpy as np
+import requests
+import model2vec
+from lancedb.embeddings import TextEmbeddingFunction, get_registry, register
+from lancedb.pydantic import LanceModel, Vector
+from chonkie import RecursiveChunker, BaseChunker, CodeChunker
+from requests.exceptions import HTTPError
+
+# TODO figure out a mechanism to retrieve the user's dlt version
+# and set it at server init.
+DLT_VERSION = "1.18.1"
+DLT_DOCS_RELATIVE_PATH = ".dlt/mcp"
+DLT_DOCS_TABLE_NAME = "pages"
+DLT_DOCS_CHUNKS_TABLE_NAME = "page_chunks"
+DLT_CODE_CHUNKS_TABLE_NAME = "code_chunks"
+_DOCS_PATH_SEGMENT = "docs/website/docs/"
+_CODE_PATH_SEGMENT = "dlt/"
+_GITHUB_REPOSITORY_URL = "https://github.com/dlt-hub/dlt"
+_GITHUB_REPOSITORY_API_URL = "https://api.github.com/repos/dlt-hub/dlt"
+
+
+chonkie.logger.disable_logging()
+
+
+@register("model2vec")
+class Model2VecEmbeddings(TextEmbeddingFunction):
+    name: str = "minishlab/potion-base-32M"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._embedding_model = None
+
+    def generate_embeddings(self, texts: list[str]) -> list[np.ndarray]:
+        vectors = self._load_model().encode(
```

Comment:
call it during __init__ once only else it would be loading it every time generate is called or is that by choice because I can see you calling it only once but in the future that might change example we might parallelize the embedding process